continuum/scenarios/permutations.py

```diff
@@ -0,0 +1,72 @@
+from typing import Callable, List
+
+import torch
+
+from continuum.datasets import _ContinuumDataset
+from continuum.scenarios import TransformationIncremental
+
+from torchvision import transforms
+
+
+class Permutations(TransformationIncremental):
+    """Continual Loader, generating datasets for the consecutive tasks.
+
+    Scenario: Permutations scenarios, use same data for all task but with pixels permuted.
+    Each task get a specific permutation, such as all tasks are different.
+
+    :param cl_dataset: A continual dataset.
+    :param nb_tasks: number of tasks in the continuum.
+    :param base_transformations: List of transformation to apply to all tasks.
+    :param seed: initialization seed for the permutations.
+    """
+
+    def __init__(
+            self,
+            cl_dataset: _ContinuumDataset,
+            nb_tasks: int,
+            base_transformations: List[Callable] = None,
+            seed=0
+    ):
+        list_transformations = []
+        g_cpu = torch.Generator()
+        list_seed = torch.randperm(1000, generator=g_cpu)[:nb_tasks]
+        self.seed = seed
+
+        # first task is not permuted
+        list_seed[0] = 0
+
+        for s_ in list_seed:
+            list_transformations.append([PermutationTransform(s_.item())])
+
+        super(Permutations, self).__init__(cl_dataset=cl_dataset,
+                                           nb_tasks=nb_tasks,
+                                           incremental_transformations=list_transformations,
+                                           base_transformations=base_transformations)
+
+    # We inverse permutation is after self.trsf because it is done an torch tensor
+    def get_task_transformation(self, task_index):
+        return transforms.Compose(self.trsf.transforms + self.inc_trsf[task_index])
+
+
+class PermutationTransform:
+    """Permutation transformers.
+
+    This transformer is initialized with a seed such as same seed = same permutation.
+    Seed 0 means no permutations
```

Comment:
Don't add those special cases like this.

Reply:
You prefer a rule like, "first task has no permutation" rather than "seed 0 is no permutation".
I just want to enable the "no permutation"